content/blog/2019-12-03-posits/index.md

```diff
@@ -0,0 +1,262 @@
+extra.author = "Dietrich Geisler & Edwin Peguero"
++++
+
+Computers are incapable of representing arbitrary real numbers exactly.
+This is due to two intractable facts of real numbers:
+- Uncountably Infinite Domain: There are an infinite number of real numbers.
+- Uncountably Infinite Precision: Some real numbers require infinite precision.
+ 
+Since computers use a finite number of bits, computer architects must settle on capturing a finite number of real numbers at a finite level of precision.
+This is done by fixing a mapping between bit patterns and real numbers called a *representation*.
+A representation makes a tradeoff between the quantity of representable numbers and the level of precision.
+
+## The Floating Point Representation
+
+The *floating point* representation is the most widely used.
+Numbers are written in the form `(-1^s) * 1.m * 2^e`, where `1.m`, the *mantissa*, and `e`, the *exponent*, are fractional and integer binary values, respectively, and `s` is a single bit denoting the sign of the represented number.
+The design tradeoff between quantity and precision is captured by the number of bits dedicated to the mantissa and exponent.
+
+In practice, however, the IEEE 754 floating point standard slightly modifies this scheme to account for two perceived limitations:
+- Small Number Gap: there is a relatively large gap between the representation of the largest negative number, and the smallest positive number
+    - To account for this, the numbers with the smallest exponent are *denormalized*. 
+    Denormalized values are spread out linearly, rather than exponentially.
+    For floating points, denormalization occurs between the largest negative and smallest positive numbers raised to the second largest exponent.
+
+Before denormalization:
+![](not-denormalized.png)
+
+After denormalization:
+![](denormalized.png)
+
+- Bogus Results: the result of overflow (e.g., dividing by a very small number) or partial functions applied to elements outside of their domains (e.g, division by zero) have no representation
+    - The case of overflow is captured by the *positive and negative infinity* values, each represented by the bit pattern corresponding to an all ones exponent and all zeros mantissa, and differentiated by the sign bit.
+    - The case of a non-result of a partial function is captured by the *NaN* value (meaning, "not a number"), represented by the various bit patterns with an all ones exponent and non-zero mantissa.
+
+All this results in a glutonous representation:
+![](float-fun.png)
+
+
+## The Posit Representation
+
+The *posit representation* _should_ be the most widely used representation.
+The numbers represented by posits are similar to floating points, but differ by the introduction of a so-called *regime* term, as follows: 
+
+````(-1^s) * 1.m * useed^k * 2^e````
+
+`useed = 2^(2^es)`, a fundamental quantity in the theory of numerical representations, parametrized by the quantity `es`.
```

Comment:
Hmm, I don't quite understand this sentence… is this one of your ""jokes,"" or is this really a fundamental quantity of some kind? Also, what exactly is $es$?

Reply:
yeah, not a very good one...  just an artifact of trying to have some fun while writing; I'll remove this one.